addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationHandler.java

```diff
@@ -0,0 +1,767 @@
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link ThirdGenerationHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author René Stakemeier - Initial contribution
+ */
+@NonNullByDefault
+public class ThirdGenerationHandler extends BaseThingHandler {
+
+    private ThirdGenerationInverterTypes inverterType;
+
+    // GSON handler
+    private static final Gson GSON = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    // used for logging
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    // base URL of the web api
+    private static final String WEB_API = "/api/v1";
+
+    /*
+     * operations used for authentication
+     */
+    private static final String AUTH_START = "/auth/start";
+    private static final String AUTH_FINISH = "/auth/finish";
+    private static final String AUTH_CREATE_SESSION = "/auth/create_session";
+
+    /*
+     * operations used for gathering process data from the device
+     */
+    private static final String PROCESSDATA = "/processdata";
+
+    /*
+     * List of available channels for the concrete device
+     */
+
+    /*
+     * After the authentication the result (the session id) is stored here and used to "sign" future requests
+     */
+    @Nullable
+    protected String sessionId = null;
+    /*
+     * The configuration file containing the host, the password and the refresh interval
+     */
+    @Nullable
+    private ThirdGenerationConfiguration config;
+
+    @Nullable
+    protected ScheduledFuture<?> refreshScheduler;
+
+    /**
+     * Constructor of this class
+     *
+     * @param thing the thing
+     */
+    public ThirdGenerationHandler(Thing thing, ThirdGenerationInverterTypes inverterType) {
+        super(thing);
+        this.inverterType = inverterType;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        /*
```

Comment:
Don't use block comments inside methods. Only use line comments: `//`

Reply:
Does this styling convention only apply to single-line comments or also mutli-line comments?

Reply:
Also multi line. The idea is it's very hard to comment out parts of code when using block comments in code. Not that code should be commented out. But block comments can't be nested, while line comments can be put inside block comments.

Reply:
Ok, corrected this. will be in the next commit